### PR TITLE
predicate pushdown fix on complex expression

### PR DESCRIPTION
**Pull request by chngpe**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java [hunk 1]
```diff
@@ -19,26 +19,112 @@
...
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.amazonaws.athena.connector.lambda.domain.predicate.expression.ConstantExpression.DEFAULT_CONSTANT_EXPRESSION_BLOCK_NAME;
 
 /**
  * Based on com.amazonaws.athena.connectors.jdbc.manager.JdbcFederationExpressionParser class
  */
 public class BigQueryFederationExpressionParser extends FederationExpressionParser
```
Comment (burhan94) [inline athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java:52]: why does BQ need all of this extra logic?

Comment (chngpe) [reply athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java:52]: BQ does not used jdbc nor support prepare statement.  What team has done in the past is to create a plan BigQuery query. There is a bug on dealing with date in the existing logic

Comment (chngpe) [reply athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java:52]: we change the signature of FederationExpressionParser, hence we need to keep the old logic in BQ.

File: athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java [hunk 2]
```diff
@@ -19,26 +19,112 @@
...
+        Block values = constantExpression.getValues();
+        FieldReader fieldReader = values.getFieldReader(DEFAULT_CONSTANT_EXPRESSION_BLOCK_NAME);
+
+        List<String> constants = new ArrayList<>();
+
+        for (int i = 0; i < values.getRowCount(); i++) {
+            fieldReader.setPosition(i);
+            String strVal = BlockUtils.fieldToString(fieldReader);
+            constants.add(strVal);
+            LOGGER.error("");
```
Comment (akuzin1) [inline athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java:114]: Is this line and 119, leftover debugging comments?

**Review by akuzin1 — APPROVED** [review]

**Review by ejeffrli — APPROVED** [review]

**Pull request merged**